tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -607,30 +627,80 @@ void TangoRosNode::PublishStaticTransforms() {
 
 void TangoRosNode::OnPoseAvailable(const TangoPoseData* pose) {
   if (publisher_config_.publish_device_pose) {
-    if (pose->frame.base == TANGO_COORDINATE_FRAME_START_OF_SERVICE
-        && pose->frame.target == TANGO_COORDINATE_FRAME_DEVICE) {
+    if (framePairMatches(pose->frame, TANGO_COORDINATE_FRAME_START_OF_SERVICE,
+                         TANGO_COORDINATE_FRAME_DEVICE)) {
       if (pose->status_code == TANGO_POSE_VALID && pose_available_mutex_.try_lock()) {
-        toTransformStamped(*pose, time_offset_, &start_of_service_T_device_);
-        start_of_service_T_device_.header.frame_id =
-          toFrameId(TANGO_COORDINATE_FRAME_START_OF_SERVICE);
-        start_of_service_T_device_.child_frame_id =
-          toFrameId(TANGO_COORDINATE_FRAME_DEVICE);
-        TangoCoordinateFramePair pair;
-        pair.base = TANGO_COORDINATE_FRAME_AREA_DESCRIPTION;
-        pair.target = TANGO_COORDINATE_FRAME_START_OF_SERVICE;
-        TangoPoseData area_description_T_start_of_service;
-        TangoService_getPoseAtTime(0.0, pair, &area_description_T_start_of_service);
-        if (area_description_T_start_of_service.status_code == TANGO_POSE_VALID) {
-          toTransformStamped(area_description_T_start_of_service,
-                             time_offset_, &area_description_T_start_of_service_);
-          area_description_T_start_of_service_.header.frame_id =
+        if (localization_status_ == LocalizationStatus::LOCALIZING) {
+          // Localization is not used or we are not localized yet,
+          // start_of_service_T_area_description is the identity.
+          toTransformStamped(*pose, time_offset_, &area_description_T_device_);
+          area_description_T_device_.header.frame_id =
               toFrameId(TANGO_COORDINATE_FRAME_AREA_DESCRIPTION);
-          area_description_T_start_of_service_.child_frame_id =
+          area_description_T_device_.child_frame_id =
+              toFrameId(TANGO_COORDINATE_FRAME_DEVICE);
+          start_of_service_T_area_description_.header.stamp = ros::Time::now();
```

Comment:
In case of VIO, the area description frame doesn't have a meaning. Is it possible to only publish area description frame in case of `localization_mode_` equals COM or Area Learning and publish device wrt. SOS in case of VIO?